web/js/MapUI/Components/UpdateProjection/UpdateProjection.js

```diff
@@ -0,0 +1,399 @@
+      compareLayerGroups.forEach((layerGroup) => map.addLayer(layerGroup));
+      compareMapUi.create(map, compare.mode);
+    }
+    updateLayerVisibilities();
+  }
+
+  const onStopAnimation = function() {
+    const needsRefresh = activeLayers.some(({ type }) => type === 'granule' || type === 'vector');
+    if (needsRefresh) {
+      // The SELECT_DATE and STOP_ANIMATION actions happen back to back and both
+      // try to modify map layers asynchronously so we need to set a timeout to allow
+      // the updateDate() function to complete before trying to call reloadLayers() here
+      setTimeout(reloadLayers, 100);
+    }
+  };
+
+  /**
+ * When page is resized set for mobile or desktop
+ *
+ * @method onResize
+ * @static
+ *
+ * @returns {void}
+ */
+  function onResize() {
+    const map = ui.selected;
+    if (isMobile) {
+      map.removeControl(map.wv.scaleImperial);
+      map.removeControl(map.wv.scaleMetric);
+    } else {
+      map.addControl(map.wv.scaleImperial);
+      map.addControl(map.wv.scaleMetric);
+    }
+  }
+
+  /**
+ * Show Map
+ *
+ * @method showMap
+ * @static
+ *
+ * @param {object} map - Openlayers Map obj
+ *
+ * @returns {void}
+ */
+  function showMap(map) {
+    document.getElementById(`${map.getTarget()}`).style.display = 'block';
+  }
+
+  /**
+ * Hide Map
+ *
+ * @method hideMap
+ * @static
+ *
+ * @param {object} map - Openlayers Map obj
+ *
+ * @returns {void}
+ */
+  function hideMap(map) {
+    document.getElementById(`${map.getTarget()}`).style.display = 'none';
+  }
+
+
+  const updateProjection = (start) => {
+    if (ui.selected) {
+      // Keep track of center point on projection switch
+      ui.selected.previousCenter = ui.selected.center;
+      hideMap(ui.selected);
+    }
+    ui.selected = ui.proj[proj.id];
+    const map = ui.selected;
+
+    const isProjectionRotatable = proj.id !== 'geographic' && proj.id !== 'webmerc';
+    const currentRotation = isProjectionRotatable ? map.getView().getRotation() : 0;
+    const rotationStart = isProjectionRotatable ? models.map.rotation : 0;
+    const rotation = start ? rotationStart : currentRotation;
+
+    updateMapUI(ui, rotation);
+
+    reloadLayers();
+
+    // If the browser was resized, the inactive map was not notified of
+    // the event. Force the update no matter what and reposition the center
+    // using the previous value.
+    showMap(map);
+
+    map.updateSize();
+
+    if (ui.selected.previousCenter) {
+      ui.selected.setCenter(ui.selected.previousCenter);
+    }
+
+    if (start) {
+      const projId = proj.selected.id;
+      let extent = null;
+      let callback = null;
+      if (models.map.extent) {
+        extent = models.map.extent;
+      } else if (!models.map.extent && projId === 'geographic') {
+        extent = getLeadingExtent(config.pageLoadTime);
+        callback = () => {
+          const view = map.getView();
+          const extent = view.calculateExtent(map.getSize());
```

Comment:
"extent" is declared at line 310. Do we need to declare a new "extent" variable here? Or is the intention to update the existing "extent" variable?

Reply:
Yeah these lines only get called if `start` is true which represents if the projection is being loaded for the first time. So we are reassigning the value of `extent` from `null`.